src/saturn/sattypes.h

```diff
@@ -10,20 +10,25 @@ typedef unsigned short u16;
 typedef unsigned int u32;
 typedef unsigned long long u64;
 
+#define true 1
+#define false 0
+#define bool s32
+
+#define NULL 0
+
+#define STAGE_ST0 0x1F
+#define TOTAL_ENTITY_COUNT 256
+
+#define NA_SE_PL_COLLECT_HEART 0x67A
+
+#define PLAYER g_Entities[PLAYER_CHARACTER]
+
 #define LOW(x) (*(s32*)&(x))
+#define ABS(x) ((x) < 0 ? -(x) : (x))
 
-typedef void (*PfnEntityUpdate)(struct Entity*);
+struct Entity;
 
-struct Unk0600B344 {
```

Comment:
Does this not need a forward declaration since it's used in the Entity struct?

Reply:
/* 0x00 */ struct Unk0600B344* unk0;
It's forward declared right there inside the Entity struct.